test/TextEditor.test.tsx

```diff
@@ -0,0 +1,53 @@
+import { StrictMode, useState } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import DataGrid, { TextEditor } from '../src';
+import type { Column } from '../src';
+import { getCells } from './utils';
+
+interface Row {
+  readonly name: string;
+}
+
+const columns: readonly Column<Row>[] = [{ key: 'name', name: 'Name', editor: TextEditor }];
+const initialRows: readonly Row[] = [{ name: 'Tacitus Kilgore' }];
+
+function Test() {
+  const [rows, setRows] = useState(initialRows);
+
+  return <DataGrid columns={columns} rows={rows} onRowsChange={setRows} />;
+}
+
+test('TextEditor', () => {
+  render(
+    <StrictMode>
+      <Test />
+    </StrictMode>
+  );
+
+  userEvent.dblClick(getCells()[0]);
+  let input: HTMLInputElement | null = screen.getByRole('textbox') as HTMLInputElement;
+  expect(input).toHaveClass('rdg-text-editor');
+  // input value is row[column.key]
+  expect(input).toHaveValue(initialRows[0].name);
+  // input is focused
+  expect(input).toHaveFocus();
+  // input value is fully selected
+  expect(input.selectionStart).toBe(0);
+  expect(input.selectionEnd).toBe(initialRows[0].name.length);
+
+  // pressing escape closes the editor without committing
+  userEvent.keyboard('Test{escape}');
+  expect(input).not.toBeInTheDocument();
+  input = null;
+  expect(getCells()[0]).toHaveTextContent(/^Tacitus Kilgore$/);
+
+  // blurring the input closes and commits the editor
+  userEvent.dblClick(getCells()[0]);
+  input = screen.getByRole('textbox') as HTMLInputElement;
+  userEvent.keyboard('Jim Milton');
+  fireEvent.blur(input);
```

Comment:
Should we press tab 
```suggestion
  userEvent.tab();
```

or 
```suggestion
  userEvent.click(document.body);
```

Reply:
I thought about both of these, but we handle these internally in a special manner (keyboard navigation, outside click detection).
I wanted to explicatively test blurring, which can happen when you alt+tab for example.